### DISCUSSION TITLE
Is it possible to add a custom script from within a plugin?

### DISCUSSION THREAD

**Discussion by ametis70** [payloadcms/payload#9811] [Q&A] [answered]
I'm trying to write a plugin for seeding a payload instance, and I know I can use `onInit` with an env var to toggle if it runs when payload starts, but could it be possible to add a script to trigger it manually using `npx payload seed` or so?

I've found the `bin` property in the config, but from the TypeScript types, it seems that it receives an array of `{ key: string, scriptPath: string }`, and I expected it to be something more like `{key: string, script: (payload: Payload) => Promise<void> }` to match the `config.onInit` API.

I took a look at the source code and found this: https://github.com/payloadcms/payload/blob/f09ee0b84be0a62d182a6fa118d33381482bcd64/packages/payload/src/bin/index.ts#L66-L67
It seems that the config is passed to a function that is the default export for the script file, but the payload instance is not passed in any way.

Also, I'm not sure if the path passed to `scriptPath` should be relative to root or the repo, relative to plugin file or relative to the CWD.

Thanks in advance!

**Comment by r1tsuu** [comment, marked as answer]
For bin scripts this
```
 (payload: Payload) => Promise<void> 
```
shouldn't be a thing, since you don't want to introduce your bin script dependencies to user's app bundle and Payload's scripts don't work that way either.

To answer your questions:
* `scriptPath` should have an absolute path like `/users/user/my-app/node_modules/payload-plugin/script.js`. So, for your plugin you should ensure that this path also will exist for those who installed it.
* Payload instance isn't passed, but the `config` is, so you can get payload from `getPayload({ config })`.
https://github.com/payloadcms/payload/blob/f09ee0b84be0a62d182a6fa118d33381482bcd64/packages/payload/src/bin/index.ts#L66-L67

**Comment by ametis70** [comment]
Thanks, I wasn't aware of `getPayload`, that should do the trick!
